### ISSUE TITLE
Cannot send CTRL modifier when pressing on hotkey.

### ISSUE THREAD

**Issue by technohiker** [autokey/autokey#989]
### AutoKey is a Xorg application and will not function in a Wayland session. Do you use Xorg (X11) or Wayland?

Xorg

### Has this issue already been reported?

- [X] I have searched through the existing issues.

### Is this a question rather than an issue?

- [X] This is not a question.

### What type of issue is this?

Bug

### Choose one or more terms that describe this issue:

- [X] autokey triggers
- [ ] autokey-gtk
- [X] autokey-qt
- [ ] beta
- [ ] bug
- [ ] critical
- [ ] development
- [ ] documentation
- [ ] enhancement
- [ ] installation/configuration
- [ ] phrase expansion
- [ ] scripting
- [ ] technical debt
- [ ] user interface

### Which Linux distribution did you use?

Linux Mint 21.2

### Which AutoKey GUI did you use?

Qt

### Which AutoKey version did you use?

96.0

### How did you install AutoKey?

Installed the deb files in the latest release.

### Can you briefly describe the issue?

I am trying to replace my `.` key with `CTRL` specifically while playing an MMO.  This is being done with the following script.
```
# Check if FFXIV is the active window
active_window = window.get_active_title()
if "FINAL FANTASY XIV" in active_window:
    # Simulate CTRL when . is pressed
    keyboard.send_key('<ctrl>')
else:
    # Normal . key behavior when not in FFXIV
    keyboard.send_key('.')
```

This script activates when `.` is pressed as a hotkey for the script.  The active_window part works, my keys behave normally when I am not in FFXIV. However, I want to send the CTRL modifier through so I can then combine it with other keys.(`. + 1`, `. + 3`, etc.)  I wanted to do this since I use IJKL for movement and my pinky rests on `/`, and reaching out to `CTRL r` isn't comfortable.

The script does not register `CTRL` at all.  Other keys work in FFXIV like `keyboard.send_key('f')`, but CTRL, SHIFT and ALT do not seem to do anything.  Furthermore, occasionally when pressing to get that key, instead I get Nums Lock, and I don't understand why that happens.

### Can the issue be reproduced?

Always

### What are the steps to reproduce the issue?

1. Create the script that was posted.
2. Set `.` hotkey.
3. Press `.` in the desired window.

### What should have happened?

I should be able to hold down `.` and have the computer read that as holding down `CTRL`.

### What actually happened?

Either no input, or Nums Lock was inputted instead.

<br/>
<hr/>

<details><summary>This repo is using Opire - what does it mean? 👇</summary><br/>💵 Everyone can add rewards for this issue commenting <code>/reward 100</code> (replace <code>100</code> with the amount).<br/>🕵️‍♂️ If someone starts working on this issue to earn the rewards, they can comment <code>/try</code> to let everyone know!<br/>🙌 And when they open the PR, they can comment <code>/claim #989</code> either in the PR description or in a PR's comment.<br/><br/>🪙 Also, everyone can tip any user commenting <code>/tip 20 @technohiker</code> (replace <code>20</code> with the amount, and <code>@technohiker</code> with the user to tip).<br/><br/>📖 If you want to learn more, check out our <a href="https://docs.opire.dev">documentation</a>.</details>

**Comment by josephj11** [comment]
Welcome to the AutoKey community, @technohiker !

TL;DR: you can't get (directly) there from here.

The way keyboards were orginally designed was to emit codes in ASCII. Pressing the Ctrl or shift keys set something in the keyboard hardward (or later, firmware) that flipped a bit in the character code that was also being pressed or subsequently pressed while holding the modifier key. So Ctrl or shift by itself didn't send anything. These (and other newer ones) are called modifier keys becauae they don't act alone. Newer systems send hardware scan codes whenever you press anything, but most programs don't work at that low level. To access them is similar to rewritng the keyboard driver.

What that means is that no regular program will respond at all to pressing a modifier key, but it will respond to any normal key being pressed as a separate keyboard event - without waiting for another keypress.

Bottom line is that what you want to do, the way you want too do it, is impossible.  This has nothing to do with AutoKey.

However, this is Linux, so almost nothing is impossible _if_ you want it badly enough.

If you are ready to go down a rabbit hole, you can design some sort of script (in AutoKey or in bash, etc.) which remaps your keyboard using xmodmap or xkb that actually makes the period into the Ctrl key and another one that puts it back to a period. These could be called from within AutoKey.

I haven't done much with those programs, but I have a pile of URLs to sites that explore them in excruciating detail.
What I seem to remember is that there was some sort of timing issue with when keyboard remapping takes effect, but that was a lot of years ago.

If you're prepared for research and experimentation, you may be able to get something like ths to work, but it may take some effort.

**Comment by josephj11** [comment]
You may want to read our [wiki](https://github.com/autokey/autokey/wiki/Emitting-Keyboard-Events) to get a better understanding of emitting keyboard events. `keyboard.send_key()` is only used when nothing else will work.

Our wiki contains a number of useful articles and lots of example scripts.

For support or to discuss anything AutoKey related, check out our [Google Groups forum](https://groups.google.com/forum/#!forum/autokey-users), or, for more advanced topics, [Gitter](https://gitter.im/autokey/autokey).

**Comment by technohiker** [comment]
I see.  If that's the case then I do not think I will be pursuing my goal with Autokey.  Thank you for the technical breakdown on why this wouldn't work so simply.
